assets/js/leaves.js

```diff
@@ -42,28 +42,42 @@ class Leaf {
   // constant for selection classname
   static selectedClass = "select";
 
-  static deselectAll() {
-    // deselect any leaf or leaf label that is currently highlighted
-    let selected = document.getElementsByClassName(Leaf.selectedClass);
-    // convert to array rather than iterating, since htmlcollection is live
-    // and changes as updated
-    Array.from(selected).forEach((item) => {
-      item.classList.remove(Leaf.selectedClass);
-    });
-
-    // change url location hash to indicate no leaf is selected
-    // needs to be non-empty to avoid page reload
-    window.location.replace("#");
-    history.replaceState(null, "", window.location.hash);
+  static setCurrentLeaf(event) {
+    // Are we removing the leaf?
+    if (event == undefined) {
+      // remove hash
+      let urlNoHash = window.location.pathname + window.location.search;
+      history.replaceState(null, "", urlNoHash);
+    } else {
+      // get the actual leaf target from DOM
+      let target = Leaf.getLeafTarget(event.target);
+      // get leaf ID
+      let leafID = target.dataset.id;
+      // update URL to reflect the currently selected leaf;
+      // replace the location & state to avoid polluting browser history
+      window.location.replace(`#${leafID}`);
+      history.replaceState(null, "", `#${leafID}`);
```

Comment:
I think we don't actually need both of these; let's drop the `location.replace` since it doesn't work for changing tags without a reload, and then we can always update the url the same way everywhere, no matter what we're changing.